crtauth/server.py

```diff
@@ -79,15 +82,21 @@ def __init__(self, secret, key_provider, server_name, token_lifetime=60,
         self.token_lifetime = token_lifetime
         self.secret = secret
         self.key_provider = key_provider
-        self.urandom = open("/dev/urandom", "r")
-        if not re.match("^[a-zA-Z0-9.-]+$", server_name):
+
+        self.server_name = server_name
+        server_string = server_name
+        if six.PY3:
+            server_string = server_name.decode('utf-8')
```

Comment:
it seems we are calling a byte array server_string here. I think that qualifies as confusing naming. Perhaps we should just keep it as a string and convert it on use if need the bytes.